### ISSUE TITLE
Haraka crashes at outbound mail after remote party returns code 421, and closing connection

### ISSUE THREAD

**Issue by shurli** [haraka/Haraka#2222]
### system info

Haraka | Haraka.js — Version: 2.8.16
 --- | :---
Node | v6.11.4
OS | Linux mailgateway 4.4.0-97-generic #120-Ubuntu SMP Tue Sep 19 17:28:18 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 1.0.2g  1 Mar 2016

### Observed behavior
Haraka crashes with sending outbound mail after return code 421, and other server closing connection.

Log:

```
2017-10-18T13:14:54.666Z [DEBUG] [-] [core] Temp fail for: Upstream error: 421 Enter message, ending with "." on a line by itself mx1.bon.at SMTP incoming data timeout - closing connection.
2017-10-18T13:14:54.666Z [DEBUG] [D5B3B18F-A74F-47B6-8202-5A73BE021593.1.1] [outbound] running deferred hooks
2017-10-18T13:14:54.677Z [DEBUG] [-] [core] [outbound] release_client: 213.33.87.13:25 to undefined
2017-10-18T13:14:54.678Z [DEBUG] [-] [core] [outbound] destroying pool entry for 213.33.87.13:25
2017-10-18T13:14:54.683Z [CRIT] [-] [core] Error: This socket has been ended by the other party
2017-10-18T13:14:54.683Z [CRIT] [-] [core]     at TLSSocket.writeAfterFIN [as write] (net.js:290:12)
2017-10-18T13:14:54.684Z [CRIT] [-] [core]     at pluggableStream.stream.Stream.pluggableStream.write (/usr/lib/node_modules/Haraka/tls_socket.js:112:34)
2017-10-18T13:14:54.684Z [CRIT] [-] [core]     at ReadStream.ondata (_stream_readable.js:555:20)
2017-10-18T13:14:54.684Z [CRIT] [-] [core]     at emitOne (events.js:101:20)
2017-10-18T13:14:54.684Z [CRIT] [-] [core]     at ReadStream.emit (events.js:188:7)
2017-10-18T13:14:54.684Z [CRIT] [-] [core]     at readableAddChunk (_stream_readable.js:176:18)
2017-10-18T13:14:54.684Z [CRIT] [-] [core]     at ReadStream.Readable.push (_stream_readable.js:134:10)
2017-10-18T13:14:54.684Z [CRIT] [-] [core]     at onread (fs.js:2018:12)
2017-10-18T13:14:54.685Z [CRIT] [-] [core]     at FSReqWrap.wrapper [as oncomplete] (fs.js:683:17)
2017-10-18T13:14:54.686Z [NOTICE] [-] [core] Shutting down
```

### Steps to reproduce
pools are disabled in outbound.ini

```ini
pool_timeout=0
pool_concurrency_max=0
```

but the behaviour is the same with the default values of the upper named two settings. (pools enabled)
also the same with disabled TLS.

```
Pakets captured at first hop after haraka (router)
Pk Num  rel.Time       Sender          Receiver        Proto     Size   Summary
      1 0.000000       *haraka*        *other*          TCP      74     24598 → 25 [SYN] Seq=0 Win=29200 Len=0 MSS=1460 SACK_PERM=1 TSval=5027717 TSecr=0 WS=128
      2 0.009654       *other*          *haraka*        TCP      60     25 → 24598 [SYN, ACK] Seq=0 Ack=1 Win=29200 Len=0 MSS=1380
      3 0.009970       *haraka*        *other*          TCP      54     24598 → 25 [ACK] Seq=1 Ack=1 Win=29200 Len=0
      4 0.020702       *other*          *haraka*        SMTP     113    S: 220 XXXXXXXXX ESMTP Exim Wed, 18 Oct 2017 21:12:48 +0200
      5 0.021132       *haraka*        *other*          TCP      54     24598 → 25 [ACK] Seq=1 Ack=60 Win=29200 Len=0
      6 0.023550       *haraka*        *other*          SMTP     81     C: EHLO XXXXXXXXX
      7 0.032570       *other*          *haraka*        SMTP     188    S: 250 XXXXXXXXX Hello XXXXXXXXX [XXXXXXXXX] | 250 SIZE 104857600 | 250 8BITMIME | 250 PIPELINING | 250 STARTTLS | 250 HELP
      8 0.034500       *haraka*        *other*          SMTP     131    C: MAIL FROM:<XXXXXX@XXX>
      9 0.049117       *other*          *haraka*        SMTP     62     S: 250 OK
     10 0.049701       *haraka*        *other*          SMTP     97     C: RCPT TO:<XXXXXX@XXX>
     11 0.098583       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=202 Ack=148 Win=29200 Len=0
     12 0.113638       *other*          *haraka*        SMTP     68     S: 250 Accepted
     13 0.114322       *haraka*        *other*          SMTP     60     C: DATA
     14 0.122767       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=216 Ack=154 Win=29200 Len=0
     15 0.123060       *other*          *haraka*        SMTP     110    S: 354 Enter message, ending with "." on a line by itself
     16 0.125682       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     17 0.125722       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     18 0.125955       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     19 0.125991       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     20 0.126018       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     21 0.126062       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     22 0.126089       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     23 0.126115       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     24 0.126142       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     25 0.126169       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     26 0.136551       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=2914 Win=34500 Len=0
     27 0.136858       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     28 0.136892       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     29 0.136999       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     30 0.137040       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     31 0.139084       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=5674 Win=40020 Len=0
     32 0.139397       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     33 0.139427       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     34 0.139485       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     35 0.139512       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     36 0.141304       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=8434 Win=45540 Len=0
     37 0.141688       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     38 0.141734       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     39 0.141762       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
     40 0.141790       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes

long time ( ~ 0,7s)  DATA fragments and TCP Acks

    961 0.812260       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    962 0.812294       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    963 0.814270       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=830914 Win=65535 Len=0
    964 0.814668       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    965 0.816275       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=833674 Win=65535 Len=0
    966 0.816553       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    967 0.816599       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    968 0.819003       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=836434 Win=65535 Len=0
    969 0.819273       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    970 0.819307       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    971 0.821007       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=839194 Win=65535 Len=0
    972 0.821588       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    973 0.821626       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    974 0.823253       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=841954 Win=65535 Len=0
    975 0.823488       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    976 0.823535       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    977 0.824608       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    978 0.824648       *haraka*        *other*          SMTP     729    [TCP Window Full] C: DATA fragment, 675 bytes
    979 0.825764       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=844714 Win=65535 Len=0
    980 0.826010       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    981 0.826073       *haraka*        *other*          SMTP     1434   [TCP Window Full] C: DATA fragment, 1380 bytes
    982 0.828042       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=847474 Win=65535 Len=0
    983 0.828418       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    984 0.828466       *haraka*        *other*          SMTP     1434   [TCP Window Full] C: DATA fragment, 1380 bytes
    985 0.830970       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=850234 Win=65535 Len=0
    986 0.831188       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    987 0.831222       *haraka*        *other*          SMTP     1434   [TCP Window Full] C: DATA fragment, 1380 bytes
    988 0.832756       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=852994 Win=65535 Len=0
    989 0.833178       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    990 0.833225       *haraka*        *other*          SMTP     1434   [TCP Window Full] C: DATA fragment, 1380 bytes
    991 0.834738       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=855754 Win=65535 Len=0
    992 0.834980       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    993 0.835050       *haraka*        *other*          SMTP     1434   [TCP Window Full] C: DATA fragment, 1380 bytes
    994 0.837250       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=858514 Win=65535 Len=0
    995 0.837465       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    996 0.837499       *haraka*        *other*          SMTP     1434   [TCP Window Full] C: DATA fragment, 1380 bytes
    997 0.839524       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=861274 Win=65535 Len=0
    998 0.839764       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
    999 0.839805       *haraka*        *other*          SMTP     759    C: DATA fragment, 705 bytes
   1000 0.841519       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=864034 Win=65535 Len=0
   1001 0.841778       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1002 0.841824       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1003 0.843994       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=866794 Win=65535 Len=0
   1004 0.844499       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1005 0.844536       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1006 0.846265       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=869554 Win=65535 Len=0
   1007 0.846512       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1008 0.846540       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1009 0.848749       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=872314 Win=65535 Len=0
   1010 0.849017       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1011 0.849070       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1012 0.850765       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=875074 Win=65535 Len=0
   1013 0.851176       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1014 0.851223       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1015 0.852995       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=877834 Win=65535 Len=0
   1016 0.853249       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1017 0.853290       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1018 0.855525       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=880594 Win=65535 Len=0
   1019 0.855762       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1020 0.855809       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1021 0.857507       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=883354 Win=65535 Len=0
   1022 0.857766       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1023 0.857818       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1024 0.859268       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=886114 Win=65535 Len=0

long time ( ~ 0,2s)  DATA fragments and TCP Acks

   1345 1.098487       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1346 1.098521       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1347 1.100763       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=1180054 Win=65535 Len=0
   1348 1.100991       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1349 1.102751       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=1182814 Win=65535 Len=0
   1350 1.102964       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1351 1.102994       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1352 1.104987       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=1185574 Win=65535 Len=0
   1353 1.105241       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1354 1.105268       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1355 1.107500       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=1188334 Win=65535 Len=0
   1356 1.107752       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1357 1.107800       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1358 1.109249       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=1191094 Win=65535 Len=0
   1359 1.109487       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1360 1.109518       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1361 1.111292       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=1193854 Win=65535 Len=0
   1362 1.111531       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1363 1.111558       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1364 1.114575       *other*          *haraka*        TCP      60     25 → 24598 [ACK] Seq=272 Ack=1196614 Win=65535 Len=0
   1365 1.114851       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1366 1.114890       *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes

here is something faulty, see timestamp
connection seems to be broken, but Error 421 at paket 1372 ist received normally

   1367 1.368556       *haraka*        *other*          TCP      1434   [TCP Retransmission] 24598 → 25 [ACK] Seq=1196614 Ack=272 Win=30016 Len=1380
   1368 1.880551       *haraka*        *other*          TCP      1434   [TCP Retransmission] 24598 → 25 [ACK] Seq=1196614 Ack=272 Win=30016 Len=1380
   1369 2.908484       *haraka*        *other*          TCP      1434   [TCP Retransmission] 24598 → 25 [ACK] Seq=1196614 Ack=272 Win=30016 Len=1380
   1370 4.960526       *haraka*        *other*          TCP      1434   [TCP Retransmission] 24598 → 25 [ACK] Seq=1196614 Ack=272 Win=30016 Len=1380
   1371 9.064567       *haraka*        *other*          TCP      1434   [TCP Retransmission] 24598 → 25 [ACK] Seq=1196614 Ack=272 Win=30016 Len=1380
   1372 13.166697      *other*          *haraka*        SMTP     119    S: 421 XXXXXXXXX SMTP incoming data timeout - closing connection.
   1373 13.167194      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1374 13.167243      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1375 13.167273      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1376 13.167300      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1377 13.167382      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1378 13.167409      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1379 13.167463      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1380 13.167514      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1381 13.167884      *other*          *haraka*        TCP      60     25 → 24598 [FIN, ACK] Seq=337 Ack=1260094 Win=65535 Len=0
   1382 13.168108      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1383 13.168149      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1384 13.168205      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1385 13.168231      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1386 13.168259      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1387 13.168399      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1388 13.168431      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1389 13.168469      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1390 13.168498      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1391 13.171404      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1392 13.171443      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1393 13.171473      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1394 13.171501      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1395 13.172143      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1396 13.172176      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1397 13.172335      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1398 13.172364      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1399 13.172396      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1400 13.172422      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1401 13.172449      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1402 13.172496      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1403 13.172523      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1404 13.172554      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1405 13.172608      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1406 13.172636      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1407 13.173443      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1408 13.173482      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1409 13.173532      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1410 13.173560      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1411 13.173587      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1412 13.175384      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1413 13.175411      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1414 13.175461      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1415 13.175488      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1416 13.176079      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1417 13.176160      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1418 13.176187      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1419 13.176214      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1420 13.176430      *haraka*        *other*          SMTP     1434   C: DATA fragment, 1380 bytes
   1421 13.176459      *haraka*        *other*          SMTP     729    [TCP Window Full] C: DATA fragment, 675 bytes
   1422 13.177033      *other*          *haraka*        TCP      60     25 → 24598 [RST] Seq=337 Win=0 Len=0
   1423 21.384521      *haraka*        *other*          SMTP     1434   [TCP Spurious Retransmission] C: DATA fragment, 1380 bytes
   2190 37.800493      *haraka*        *other*          SMTP     1434   [TCP Spurious Retransmission] C: DATA fragment, 1380 bytes
   3651 70.632519      *haraka*        *other*          SMTP     1434   [TCP Spurious Retransmission] C: DATA fragment, 1380 bytes
```

**Comment by shurli** [comment]
It seem that we have strange issues with paket loss on our connection. At the moment we check this with our ISP.
We tried with our fallback internet line and have no issues here.

But also if pakets are strange in this case, this should not make haraka crash

**Comment by baudehlo** [comment]
I agree it shouldn't crash anything.

**Comment by msimerson** [comment]
Probably fixed by #1603
